AEPTarget/Tests/TestHelpers/UserDefaults+clear.swift

```diff
@@ -12,10 +12,20 @@
 import Foundation
 
 extension UserDefaults {
-    static func clear() {
-        for _ in 0 ... 5 {
-            for key in UserDefaults.standard.dictionaryRepresentation().keys {
-                UserDefaults.standard.removeObject(forKey: key)
+    static func clear(appGroup: String? = nil) {
+        if let appGroup = appGroup,
+           !appGroup.isEmpty,
+           let userDefaults = UserDefaults(suiteName: appGroup) {
+                for _ in 0 ... 5 {
```

Comment:
do we have a better way to iterate through defaults to clear them?  curious as to why we are using a for loop here.

Reply:
This code has been around for a bit to ensure the defaults get cleared. You can find it elsewhere too, see https://github.com/adobe/aepsdk-core-ios/blob/main/AEPCore/Mocks/UserDefaults%2BClear.swift.